File: A2_q3.py
with open('M.txt', 'r') as f:
    M = [[float(num) for num in line.split(' ')] for line in f]
print('M =', M)

with open('N.txt', 'r') as f:
    N = [[float(num) for num in line.split(' ')] for line in f]
print('N =',N)

A=[2,3,5]
print('A =',A)
p1=[[0,0,0], [0,0,0], [0,0,0]]
p2=[0,0,0]

for i in range(3):
    for x in range(3):
        for j in range(3):
            p1[i][j] = p1[i][j] + (M[i][x] * N[x][j]) 
print("Product of M*N =", p1)

for i in range(3):
	for j in range(3):
		p2[i] = p2[i] + (M[i][j] * A[j])		
print("Product of M*A =",p2)

#Output
#M = [[4.0, -2.0, 5.0], [1.0, 3.0, -6.0], [0.0, -1.0, 7.0]]
#N = [[1.0, 6.0, -2.0], [0.0, 3.0, 4.0], [-6.0, 5.0, 7.0]]
#A = [2, 3, 5]
#Product of M*N = [[-26.0, 43.0, 19.0], [37.0, -15.0, -32.0], [-42.0, 32.0, 45.0]]
#Product of M*A = [27.0, -19.0, 32.0]

#[Program finished]

File: A2_q3.cxx
#include<iostream>
#include<fstream>
using namespace std;

int main(){
//Define M , N , M*N = P, M*A = Q
int M[3][3], N[3][3], p1[3][3], p2[3][1];

//Reading M.txt and printing
cout<<"M = "<<endl;
ifstream fm("M.txt");
for (int i = 0; i < 3; i++){
	for (int j = 0; j < 3; j++){
		fm >> M[i][j];
		cout<<M[i][j]<<" ";
	}cout<<endl;
}cout<<endl;

//Reading N.txt and printing
cout<<"N = "<<endl;
ifstream n("N.txt");
for (int i = 0; i < 3; i++){
	for (int j = 0; j < 3; j++){
		n >> N[i][j];
		cout<<N[i][j]<<" ";
	}cout<<endl;
}cout<<endl;

//Defing  vector A and printing
int A[3][1] = {2,3,5};
cout<<"A = "<<endl;
for(int i =0; i<3; i++){
	cout<<A[i][0]<<endl;
}cout<<endl;

//for M*N
cout<<"M*N = "<<endl;
for(int i=0; i<3; i++){
	for(int j=0; j<3; j++){
		for(int k=0; k<3; k++){
			p1[i][j]+=M[i][k]*N[k][j];
		}
	cout<<p1[i][j]<<" ";
	}
	cout<<endl;
}cout<<endl;

//for M*A
cout<<"M*A = "<<endl;
for(int i=0; i<3; i++){
	for(int j=0; j<3; j++){
		p2[i][0]+=M[i][j]*A[j][0];
	}cout<<p2[i][0]<<endl;
}
return 0;	
}


/*
output
M =
4 -2 5
1 3 -6
0 -1 7

N =
1 6 -2
0 3 4
-6 5 7

A =
2
3
5

M*N =
-26 43 19
37 -15 -32
-42 32 45

M*A =
27
-19
32

[Program finished]
*/